src/Microsoft.AspNetCore.Sockets.Http/HttpConnectionDispatcher.cs

```diff
@@ -314,19 +315,45 @@ private Task ProcessNegotiate(HttpContext context, HttpSocketOptions options)
             // Set the allowed headers for this resource
             context.Response.Headers.AppendCommaSeparatedValues("Allow", "GET", "POST", "OPTIONS");
 
-            context.Response.ContentType = "text/plain";
+            context.Response.ContentType = "application/json";
 
             // Establish the connection
             var connection = _manager.CreateConnection();
 
             // Get the bytes for the connection id
-            var connectionIdBuffer = Encoding.UTF8.GetBytes(connection.ConnectionId);
+            var connectionIdBuffer = Encoding.UTF8.GetBytes(GetNegotiatePayload(connection.ConnectionId, options));
 
             // Write it out to the response with the right content length
             context.Response.ContentLength = connectionIdBuffer.Length;
             return context.Response.Body.WriteAsync(connectionIdBuffer, 0, connectionIdBuffer.Length);
         }
 
+        private static string GetNegotiatePayload(string connectionId, HttpSocketOptions options)
+        {
+            var sb = new StringBuilder();
```

Comment:
Avoid the dependency + it's faster. I can use JsonTextWriter if you like it better (one of the reasons why it's in a separate method - easy to replace the implementation without changing any other code)